medical_prescription_merge/wizards/medical_prescription_order_merge.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import itertools
+from openerp import _, api, fields, models
+from openerp.exceptions import ValidationError
+
+
+class MedicalPrescriptionOrderMerge(models.TransientModel):
+    _name = 'medical.prescription.order.merge'
+
+    merge_order_ids = fields.Many2many(
+        comodel_name='medical.prescription.order',
+        string='Orders to be Merged',
+        help='List of prescription orders to be merged',
+        required=True,
+        default=lambda s: s._compute_default_merge_orders(),
+    )
+    dest_order_id = fields.Many2one(
+        comodel_name='medical.prescription.order',
+        string='Destination Order',
+        help='This prescription order will serve as the baseline for the merge'
+             ' (its info will take precedence when there are conflicts)',
+        required=True,
+        default=lambda s: s._compute_default_dest_order(),
+        domain=lambda s: s._compute_dest_order_domain(),
+    )
+    skip_validation = fields.Boolean(
+        help='Check this box to allow prescription orders issued by different'
+             ' physicians or for different dates to be merged',
+        string='Skip Validation?',
+    )
+
+    @api.model
+    def _compute_default_merge_orders(self):
+        if self.env.context.get('active_model') \
+                == 'medical.prescription.order':
+            return self.env['medical.prescription.order'].browse(
+                self.env.context.get('active_ids')
+            )
+        else:
+            return self.env['medical.prescription.order']
+
+    @api.model
+    def _compute_default_dest_order(self):
+        merge_order_ids = self._compute_default_merge_orders()
+        if merge_order_ids:
+            return merge_order_ids[0]
+        else:
+            return merge_order_ids
+
+    @api.multi
+    def _compute_dest_order_domain(self):
+        try:
+            self.ensure_one()
+            merge_ids = self.merge_order_ids.ids
+        except AssertionError:
+            merge_ids = self._compute_default_merge_orders().ids
+        return "[('id', 'in', %r)]" % merge_ids
+
+    @api.multi
+    def action_merge(self):
+        self.ensure_one()
+
+        if len(self.merge_order_ids) < 2:
+            raise ValidationError(_(
+                'You must select at least two orders to start a merge.'
+            ))
+
+        if not self.skip_validation:
+            merge_physicians = set(self.merge_order_ids.mapped('physician_id'))
```

Comment:
Move this into a method of its own `_do_validation` or similar. Allows for inherited modules to inject logic. Rule of thumb, logic in loops is inaccessible to downstream modules. This definitely should be accessible.